benchmarks/event-circuit-speed/index.js

```diff
@@ -0,0 +1,160 @@
+        downloading: ['Load testing progress'],
+        finished: 'Load testing done',
+        error: 'Load testing error'
+    }
+});
+
+function reporterHandler() {
+    if (totalEventReported.value !== totalEventProcessed.value) {
+        const tickSize = totalEventProcessed.value - totalEventReported.value;
+        progress.tick(tickSize);
+        totalEventReported.value = totalEventProcessed.value;
+    }
+
+    setTimeout(reporterHandler, 500);
+}
+
+setTimeout(reporterHandler, 500);
+
+function runLoadTests() {
+    const initialTime = new Date();
+    const child = spawn('babel-node', [path.join(__dirname, './buildState')]);
+
+    let resolver;
+    let rejecter;
+
+    const done = new Promise((resolve, reject) => {
+        resolver = resolve;
+        rejecter = reject;
+    });
+
+    const logHandler = (rawData) => {
+        const data = rawData.toString('utf8');
+
+        if (data.indexOf(DONE_TOKEN) >= 0) {
+            const transferPayload = JSON.parse(data.substring(DONE_TOKEN.length));
+            const buildTime = new Date() - initialTime;
+            child.kill();
+
+            resolver(Object.assign({ buildTime }, transferPayload));
+        } else if (data.indexOf(ERR_TOKEN) >= 0) {
+            child.kill();
+            rejecter(data.substring(ERR_TOKEN.length));
+        } else if (data.indexOf(INFO_TOKEN) >= 0) {
+            const transferString = data.substring(INFO_TOKEN.length);
+            const progressValue = parseInt(transferString, 10);
+            totalEventProcessed.value += progressValue;
+        }
+    };
+
+    child.stdout.on('data', logHandler);
+    child.stderr.on('data', logHandler);
+
+    return done;
+}
+
+function dropCollection() {
+    return MongoClient.connect(config.MONGODB_CONNECTION_URL).then(db =>
+        new Promise(resolve => db
+            .collection(config.MONGODB_COLLECTION_NAME, { strict: true }, (err, collection) => (
+                err ? resolve() : collection.drop().then(resolve)
+        )))
+        .catch(err => log('Error while drop collection:', err))
+        .then(() => db.close())
+    );
+}
+
+function generateEvents(arr, i, storage) {
+    return dropCollection()
+        .then(() => prepareEvents(arr[i], totalEventProcessed))
+        .then(runLoadTests)
+        .then((data) => {
+            storage[arr[i]] = data;
+            if (i === arr.length - 1) {
+                return storage;
+            }
+            return generateEvents(arr, i + 1, storage);
+        });
+}
+
+function fsWriteFileAsync(filename, content) {
+    return new Promise((resolve, reject) =>
+        fs.writeFile(filename, content, (err, value) => (
+            !err ? resolve(value) : reject(err)
+        ))
+    );
+}
+
+function storeToCsv(filePath, points, data, dataGetter) {
+    return fsWriteFileAsync(filePath,
+        `${points.map(point => `Events count: ${point}`).join(',')}\n` +
+        `${points.map(point => dataGetter(data, point)).join(',')}\n`
```

Comment:
`dataGetter` and `data` are used only once and together - there should be one function argument (rather then two) - `dataGetter` (which accepts one argument - `point`).